mmv1/third_party/terraform/data_sources/data_source_google_compute_addresses.go.erb

```diff
@@ -0,0 +1,205 @@
+<% autogen_exception -%>
+package google
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+<% if version == "ga" -%>
+	"google.golang.org/api/compute/v1"
+<% else -%>
+	compute "google.golang.org/api/compute/v0.beta"
+<% end -%>
+)
+
+func dataSourceGoogleComputeAddresses() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceGoogleComputeAddressesRead,
+
+		Schema: map[string]*schema.Schema{
+			"addresses": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The name of the IP address.`,
+						},
+						"address": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The IP address.`,
+						},
+						"address_type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The IP address type.`,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"region": {
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+						},
+						"self_link": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+<% unless version == 'ga' -%>
```

Comment:
Is labels beta-only? The tests include labels for addresses, will this cause problems?

Reply:
Labels for compute addresses are still in beta yes, yes I forgot to condition labels in tests resource, I'll fix thanks!